src/main/java/org/klodnicki/service/AccountService.java

```diff
@@ -47,13 +47,114 @@ private List<Account> getAllAccountsFromDatabase() {
-        if (!accountExist(pesel)) {
+        if (accountNotExist(pesel)) {
             throw new NotFoundInDatabaseException(Account.class);
         }
         accountRepository.removeAccount(pesel);
     }
 
-    private boolean accountExist(String pesel) {
-        return accountRepository.findAccountByPesel(pesel).isPresent();
+    private boolean accountNotExist(String pesel) {
+        return accountRepository.findAccountByPesel(pesel).isEmpty();
+    }
+
+    public void modifyAccount(String peselAccountToModify, String parameterToModify, String newData)
+            throws IllegalArgumentException, NotFoundInDatabaseException {
+
+        if (accountNotExist(peselAccountToModify)) {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        if (parameterToModify.equalsIgnoreCase(SortOptionAccount.PESEL.getSortName()) && newData.length() != 11) {
+            throw new IllegalArgumentException("Pesel must have 11 digits.");
+
+        }
+
+        accountRepository.mergeAccount(peselAccountToModify, prepareParameterToDatabase(parameterToModify), newData);
+    }
+
+    private String prepareParameterToDatabase(String parameter) {
+
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            if (sortOption.getSortName().equalsIgnoreCase(parameter)) {
+                parameter = sortOption.getHqlParameter();
+            }
+        }
+        return parameter;
+    }
+
+    public List<String> sortOptionNames() {
+        List<String> namesList = new ArrayList<>();
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            namesList.add(sortOption.getSortName());
+        }
+        return namesList;
+    }
+
+    public List<String> prepareListOfAccountsByFirstNameAndLastName(String firstName, String lastName)
+            throws NotFoundInDatabaseException {
+        List<Account> accountsByFirstNameAndLastName = accountRepository.findAccountsByFirstNameAndLastName
+                (firstName, lastName);
+        List<String> results = new ArrayList<>();
+
+        if (accountsByFirstNameAndLastName.isEmpty()) {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        for (Account account : accountsByFirstNameAndLastName) {
+            results.add(account.toString());
+        }
+        return results;
+    }
+
+    public List<String> prepareAllAccountsOrderByParameter(String parameter) {
+        List<Account> accountsOrderByParameter = accountRepository.findAllAccountsOrderByParameter
+                (prepareParameterToDatabase(parameter));
+        List<String> results = new ArrayList<>();
+
+        for (int i = 0; i < accountsOrderByParameter.size(); i++) {
+            results.add(accountsOrderByParameter.get(i).toString());
+        }
+        return results;
+    }
+
+    private boolean peselExist(String pesel, String firstName, String lastName) {
```

Comment:
This method is used only in one place. It can take a list as parameter and don't need to lookup database records for that as you already do it before in method which calls this one.